### DISCUSSION TITLE
How do I add the metadata to my project based on the cell barcode?

### DISCUSSION THREAD

**Discussion by zahramesri** [GreenleafLab/ArchR#1437] [Questions / Documentation]
Hello,

I have created my ArchRProject() and now I would like to add my metadata (Patient number, treatment, etc.). These information have been provided to me based on the cell barcode in an excel. For example:

cell barcode = AAACGAAAGAAGCCTG.Pre_P012_t, ...
patient = P012, ...
sample = Pre_P012_t, ...
group = pre treatment, ...

Here is a sample of head(proj$cellNames)
"scATAC#TCTAGTTTCTACTTTG-4" "scATAC#TGCGTAACACAGCTTA-1"

How do I add the metadata to my project based on the cell barcode? I could not find my answer in "[Adding data to an ArchRProject](https://www.archrproject.com/bookdown/manipulating-an-archrproject.html)" 

Any help would be appreciated. 

Best,
Zahra

**Comment by rcorces** [comment]
The answer is, in fact, in the section you linked. You will have to manipulate the data in your excel file so that you can match the cell barcode with the corresponding `cellName` and then use `addCellColData()` https://www.archrproject.com/reference/addCellColData.html
